compiler/src/iree/compiler/Dialect/LinalgExt/Utils/Utils.cpp

```diff
@@ -891,6 +891,15 @@ bool isArgmaxOp(linalg::GenericOp genericOp) {
     if (!matchPattern(producer, m_Op<arith::SelectOp>())) {
       return false;
     }
+    auto selectOp = dyn_cast<arith::SelectOp>(producerOutput.getDefiningOp());
```

Comment:
Use `cast` if you require the result to be a select. Otherwise check if the result is not null.